src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/ScannedDocumentsParser.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ScannedDocument;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.services.exception.CcdDataParseException;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ExceptionRecordFieldNames.SCANNED_DOCUMENTS;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors.errors;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors.result;
+
+/**
+ * Converts raw exception record data representing scanned documents
+ * (in the form of a list of maps of objects) into a strongly typed collection.
+ */
+@Service
+public class ScannedDocumentsParser {
+
+    private static final Logger log = LoggerFactory.getLogger(OcrDataParser.class);
+
+    private final CcdCollectionParser ccdCollectionParser;
+
+    public ScannedDocumentsParser(CcdCollectionParser ccdCollectionParser) {
+        this.ccdCollectionParser = ccdCollectionParser;
+    }
+
+    public ResultOrErrors<List<CcdCollectionElement<ScannedDocument>>> parseScannedDocuments(
+        Map<String, Object> exceptionRecordData
+    ) {
+        Object scannedDocumentsElement = exceptionRecordData.get(SCANNED_DOCUMENTS);
+
+        if (scannedDocumentsElement == null) {
+            return result(emptyList());
+        } else {
+            try {
+                List<CcdCollectionElement<ScannedDocument>> scannedDocuments =
+                    ccdCollectionParser.parseCcdCollection(scannedDocumentsElement, ScannedDocument.class);
+
+                return result(scannedDocuments);
+            } catch (CcdDataParseException e) {
+                log.warn("Failed to retrieve scanned documents from exception record", e);
+                return errors("Scanned documents collection has invalid format");
```

Comment:
This message will be shown to the Caseworker so if there are multiple scanned documents this message may not be that helpful. I am not sure but what will happen if we add exception message to this string or can the parser be nice enough to tell us which fields failed ?

Reply:
This is a situation where the service got a malformed request. Maybe we should throw an exception, because even if all other fields are valid, there's something seriously wrong with the request. There's nothing a caseworker can fix, so they should just get a general response. I'm wondering if we should just throw an exception (resulting with a 500 response) of maybe even the library should perform this check first.

Reply:
Any thoughts?

Reply:
Throwing an exception instead, in order to return a 500 response, as this is a proper error somewhere between services.

Reply:
this is done now